src/components/WarningAcknowledgement/WarningAcknowledgement.tsx

```diff
@@ -0,0 +1,104 @@
+import { Box, Button, Flex } from '@chakra-ui/react'
+import { motion } from 'framer-motion'
+import React, { useCallback } from 'react'
+import { FiAlertTriangle } from 'react-icons/fi'
+import { useTranslate } from 'react-polyglot'
+import { RawText, Text } from 'components/Text'
+
+type WarningAcknowledgementProps = {
+  children: React.ReactNode
+  message: string
+  onAcknowledge: () => void
+  shouldShowWarningAcknowledgement: boolean
+  setShouldShowWarningAcknowledgement: (shouldShow: boolean) => void
+}
+
+const cancelHoverProps = { bg: 'rgba(255, 255, 255, 0.2)' }
+const understandHoverProps = { bg: 'red.600' }
+const boxBorderRadius = { base: 'none', md: 'xl' }
+
+export const WarningAcknowledgement = ({
+  children,
+  message,
+  onAcknowledge,
+  shouldShowWarningAcknowledgement,
+  setShouldShowWarningAcknowledgement,
+}: WarningAcknowledgementProps) => {
+  const translate = useTranslate()
+
+  const handleAcknowledge = useCallback(() => {
+    setShouldShowWarningAcknowledgement(false)
+    onAcknowledge()
+  }, [onAcknowledge, setShouldShowWarningAcknowledgement])
+
+  const handleCancel = useCallback(() => {
+    setShouldShowWarningAcknowledgement(false)
+  }, [setShouldShowWarningAcknowledgement])
+
+  return shouldShowWarningAcknowledgement ? (
+    <Box position='relative' borderRadius={boxBorderRadius} overflow='hidden'>
+      <Box
+        background='rgba(0, 0, 0, 0.5)'
+        zIndex={2}
+        position='absolute'
+        right={0}
+        bottom={0}
+        width='100%'
+        height='100%'
+      >
+        {
+          <motion.div
+            initial={{ y: '100%' }}
+            animate={{ y: 0 }}
+            transition={{ duration: 0.3 }}
+            style={{
+              position: 'absolute',
+              bottom: 0,
+              left: 0,
+              right: 0,
+              background: 'rgba(0, 0, 0, 0.8)',
+              padding: '1rem',
+              color: 'white',
+              zIndex: 10,
+            }}
```

Comment:
these guys should be constants

Reply:
@woodenfurniture since the component is only rendering children and doesn't have any lifecycle hooks in its body, should be one of the few cases where we are probably fine not to have an object as a constant, though just because we can doesn't mean we shouldn't for consistency 🙏🏽

Reply:
For sure - I've moved to constants for consistency in https://github.com/shapeshift/web/pull/6647/commits/05c292f7035995dab341ddf89c2ad7e64e3e383b.